gluten-ut/spark33/src/test/scala/org/apache/spark/sql/statistics/SparkFunctionStatistics.scala

```diff
@@ -0,0 +1,189 @@
+            .getOrCreate()
+      }
+    }
+  }
+
+  def extractQuery(examples: String): Seq[String] = {
+    examples.split("\n").map(_.trim).filter(!_.isEmpty).filter(_.startsWith("> SELECT"))
+        .map(_.replace("> SELECT", "SELECT"))
+  }
+
+  test(GlutenTestConstants.GLUTEN_TEST + "Run spark function statistics: ") {
+    initializeSession
+    val functionRegistry = spark.sessionState.functionRegistry
+    val sparkBuiltInFunctions = functionRegistry.listFunction()
+    // According to expressionsForTimestampNTZSupport in FunctionRegistry.scala,
+    // these functions are registered only for testing, not available for end users.
+    // Other functions like current_database is NOT necessarily offloaded to native.
+    val ignoreFunctions = FunctionRegistry.expressionsForTimestampNTZSupport.keySet ++
+        Seq("get_fake_app_name", "current_catalog", "current_database", "spark_partition_id",
+          "current_user", "current_timezone")
+    val supportedFunctions = new java.util.ArrayList[String]()
+    val unsupportedFunctions = new java.util.ArrayList[String]()
+    val needInspectFunctions = new java.util.ArrayList[String]()
+
+    for (func <- sparkBuiltInFunctions) {
+      val exprInfo = functionRegistry.lookupFunction(func).get
+      if (!ignoreFunctions.contains(exprInfo.getName)) {
+        val examples = extractQuery(exprInfo.getExamples)
+        if (examples.isEmpty) {
+          needInspectFunctions.add(exprInfo.getName)
+          // scalastyle:off println
+          println("## Not found examples for " + exprInfo.getName)
+          // scalastyle:on println
+        }
+        var isSupported: Boolean = true
+        breakable {
+          for (example <- examples) {
+            var executedPlan: SparkPlan = null
+            try {
+              executedPlan = spark.sql(example).queryExecution.executedPlan
+            } catch {
+              case t: Throwable =>
+                needInspectFunctions.add(exprInfo.getName)
+                // scalastyle:off println
+                println("-- Need inspect " + exprInfo.getName)
+                println(exprInfo.getExamples)
+                // scalastyle:on println
+                break
+            }
+            val hasFallbackProject = executedPlan.find(_.isInstanceOf[ProjectExec]).isDefined
+            if (hasFallbackProject) {
+              isSupported = false
+              break
+            }
+            val hasGlutenPlan = executedPlan.find(_.isInstanceOf[GlutenPlan]).isDefined
+            if (!hasGlutenPlan) {
+              isSupported = false
+              break
+            }
+            break
+          }
+        }
+        if (isSupported && !needInspectFunctions.contains(exprInfo.getName)) {
+          supportedFunctions.add(exprInfo.getName)
+        } else if (!isSupported && !needInspectFunctions.contains(exprInfo.getName)) {
+          unsupportedFunctions.add(exprInfo.getName)
```

Comment:
Is there any case meet this condition `!isSupported && needInspectFunctions.contains(exprInfo.getName)`? If not, this clause can just be like `else if (!isSupported)`.

Reply:
Just removed the extra check condition that never goes into. Thanks!